android-components/components/feature/search/src/test/java/mozilla/components/feature/search/telemetry/incontent/InContentTelemetryTest.kt

```diff
@@ -114,7 +114,7 @@ class InContentTelemetryTest {
 
     @Test
     fun `GIVEN a Baidu search WHEN trackPartnerUrlTypeMetric is called THEN emit an appropriate IN_CONTENT_SEARCH fact`() {
-        val url = "https://m.baidu.com/s?from=1000969a&word=aaa"
+        val url = "https://www.baidu.com/s?tn=monline_dg&wd=aaa"
```

Comment:
Just to note, we need to resolve the situation here before we land this. We'll either need to update the Baidu search engine, or extend search-telemetry to cover this.

I'm currently thinking about what to do here. If it starts to block you, please let me know.

Reply:
I've done some more investigations, and I think we should keep both `m.baidu.com` and use `word` as the parameter name for the search term.

On mobile, Baidu uses `word` as the search term parameter name for organic searches, regardless of it you're on `www.` or `m.`.

I think because we're keeping that, we should also keep the `m.baidu.com`.

On the remote settings front, we'll need to update `searchPageRegexp` and probably `extraAdServersRegexps` to allow for `www.` or `m.`.

For the search term parameter, that's a bit harder. `queryParamName` can only be a string, but I think we should turn it into an array. So we probably need a new field, maybe `searchTermParamNames` that can be `["wd", "word"]` and then we'll have to iterate over that array.

We would also duplicate all the existing `queryParamName` fields into `searchTermParamNames` so that new clients would only need to read the new array and older desktop clients could keep using the single version.

Note that I still need to review this with the search team, but this is the way I'm currently thinking.

Reply:
Correction: We're not recording tagged codes for Baidu currently, so this sub-test can be removed.